gulp-tasks/utils/publish-helpers.js

```diff
@@ -0,0 +1,179 @@
+    cwd: sourceCodePath,
+  });
+
+  await spawn('gulp', ['build'], {
+    cwd: sourceCodePath,
+  });
+};
+
+/*
+ * This function will create a directory with the same name as the
+ * .tar.gz file it generates. This way when the file is extracted
+ * the folder structure will be the same.
+ */
+const groupBuildFiles = async (tagName, gitBranch) => {
+  const groupedBuildFiles =
+    path.join(getBuildPath(tagName), GROUPED_BUILD_FILES);
+  const dirExists = await doesDirectoryExist(groupedBuildFiles);
+  logHelper.log(oneLine`
+    Group Build Files into
+    ${logHelper.highlight(path.relative(process.cwd(), groupedBuildFiles))}.
+  `);
+
+  if (!dirExists) {
+    await downloadGitCommit(tagName, gitBranch);
+    await buildGitCommit(tagName);
+
+    const sourceCodePath = path.join(getBuildPath(tagName), SOURCE_CODE_DIR);
+
+    const pattern = path.posix.join(
+      sourceCodePath, 'packages', '**',
+      constants.PACKAGE_BUILD_DIRNAME, constants.BROWSER_BUILD_DIRNAME,
+      '*.{js,map}');
+
+    // Copy files from the source code and move into the grouped build
+    // directory. In others, have a flat file structure of just the built files.
+    const filesToIncludeInBundle = glob.sync(pattern);
+    for (let fileToInclude of filesToIncludeInBundle) {
+      await fs.copy(
+        fileToInclude,
+        path.join(groupedBuildFiles, path.basename(fileToInclude)),
+      );
+    }
+  } else {
+    logHelper.log(`   Builds files already grouped.`);
+  }
+
+  return groupedBuildFiles;
+};
+
+const createArchive = async (tagName, fileExtension, format, options) => {
+  const archiveFilesPath = path.join(getBuildPath(tagName), 'archives');
+  const archiveFilename = `workbox-${tagName}.${fileExtension}`;
+  const outputFilePath = path.join(archiveFilesPath, archiveFilename);
+
+  logHelper.log(oneLine`
+    Creating ${format} and saving to
+    ${logHelper.highlight(path.relative(process.cwd(), outputFilePath))}.
+  `);
+
+  fs.ensureDirSync(path.dirname(outputFilePath));
+
+  const groupedBuildFiles =
+    path.join(getBuildPath(tagName), GROUPED_BUILD_FILES);
+
+  await new Promise((resolve, reject) => {
+    const writeStream = fs.createWriteStream(outputFilePath);
+    writeStream.on('close', () => resolve(outputFilePath));
+
+    const archive = archiver(format, options);
+    archive.on('error', (err) => {
+      reject(err);
+    });
+    archive.pipe(writeStream);
+    // Adds the directory contents to the zip.
+    archive.directory(groupedBuildFiles, false);
+    archive.finalize();
+  });
+
+  return outputFilePath;
+};
+
+const createTarGz = async (tagName) => {
+  return createArchive(tagName, 'tar.gz', 'tar', {
+    gzip: true,
+  });
+};
+
+const createZip = (tagName) => {
+  return createArchive(tagName, 'zip', 'zip', {
+    zlib: {level: 9},
+  });
+};
+
+const createBundles = async (tagName, gitBranch) => {
+  await groupBuildFiles(tagName, gitBranch);
+
+  const tarPath = await createTarGz(tagName);
```

Comment:
If you want to run both archive creations [in parallel](https://developers.google.com/web/fundamentals/primers/async-functions#careful_avoid_going_too_sequential):

```js
const [tarPath, zipPath] = await Promise.all([createTarGz(tagName), createZip(tagName)]);
return {tarPath, zipPath};
```

Reply:
To be honest, I want the publish process to be as sane and reduce any errors as possible so would prefer to keep this sequential just in that vain (plus it's relatively quick anyway).